packages/diagrams/lib/src/application_lifecycle.dart

```diff
@@ -0,0 +1,318 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+import '../diagrams.dart';
+
+class AppLifecycleDiagram extends StatefulWidget with DiagramMetadata {
+  const AppLifecycleDiagram({super.key, required this.name});
+
+  @override
+  final String name;
+
+  static const Color arrowBorderColor = Color(0xFF858585);
+  static const Color arrowColor = arrowBorderColor;
+  static const Color transitionLabelColor = arrowBorderColor;
+  static const double arrowThickness = 20;
+  static const double stateBoxWidth = 240;
+  static const double stateBoxHeight = 100;
+  static const double overallWidth = 1000;
+  static const double overallHeight = 360;
+  static const double middleArrowWidth = (overallWidth - 3 * stateBoxWidth) / 2;
+  static const double startArrowWidth = overallWidth - 2 * stateBoxWidth;
+  static const double verticalArrowHeight = overallHeight - 2 * stateBoxHeight;
+  static const double middleSpacerHeight = verticalArrowHeight;
+
+  @override
+  State<AppLifecycleDiagram> createState() => _DiagramState();
+}
+
+class _DiagramState extends State<AppLifecycleDiagram> {
+  @override
+  Widget build(BuildContext context) {
+    return ConstrainedBox(
+      key: UniqueKey(),
```

Comment:
Is this relevant for anything? Maybe leave a comment?

Reply:
No, not relevant, removed.